EventListener/SetEmailAsUsernameListener.php

```diff
@@ -16,26 +18,18 @@
 
 class SetEmailAsUsernameListener implements EventSubscriberInterface
 {
-    /**
-     * {@inheritdoc}
-     */
-    public static function getSubscribedEvents()
+    public static function getSubscribedEvents(): array
     {
         return [
             AdminSecurityEvents::USER_CREATED => 'setEmailAsUsername'
         ];
     }
 
-    /**
-     * @param UserEvent $event
-     *
-     * @throws \Exception
-     */
-    public function setEmailAsUsername(UserEvent $event)
+    public function setEmailAsUsername(UserEvent $event): void
     {
         $user = $event->getUser();
         if (!$user instanceof UserInterface) {
-            throw new \Exception('User entity should implement UserInterface');
```

Comment:
Same here.

Reply:
Exception is not necessary here. If `$user` does not implement `FSi\Bundle\AdminSecurityBundle\Security\User\UserInterface` then it does not have `setUsername` method. In such case the intention is probably not to copy email as username rather than throwing an error at end user.